tests/display/assoc_ty.rs

```diff
@@ -0,0 +1,305 @@
+#[test]
```

Comment:
Just my general thoughts on these tests: great coverage, but I think I would like these to be combined when possible. For example, the test below could just merge the two programs.

Addtionally, comments for *what* each test/program is for would be nice.